### PR TITLE
Add Onnx export support for string processing transformers

### PR DESCRIPTION
**Pull request by harishsk**
This draft PR adds support for exporting to Onnx, the following transformers. 
- NgramTransformer
- ValueToKeyMappingTransformer (upgraded support)
- WordTokenizingTransformer

This PR also upgrades the version of Google Protobuf library.

This work is not ready for merging because this work relies on a new version of the Onnx runtime due to be released later. I am submitting this PR for review now so that this can be merged soon after the release of the runtime. 

Please review the changes so that I can iterate on the review comments now, before the public release of the ORT.

### CODE REVIEW

File: src/Microsoft.ML.Data/Transforms/ValueToKeyMappingTransformer.cs [hunk 1]
```diff
@@ -768,22 +768,68 @@ protected override Delegate MakeGetter(DataViewRow input, int iinfo, Func<int, b
...
+                else
+                {
+                    return false;
+                }
                 node.AddAttribute("default_int64", -1);
-                //default_string needs to be an empty string but there is a BUG in Lotus that
-                //throws a validation error when default_string is empty. As a work around, set
-                //default_string to a space.
-                node.AddAttribute("default_string", " ");
+                node.AddAttribute("values_int64s", termIds);
```
Comment (yaeldekel) [inline src/Microsoft.ML.Data/Transforms/ValueToKeyMappingTransformer.cs]: >values_int64s [](start = 35, length = 13)

I'm not very familiar with ONNX - aren't the `termIds` the keys and `terms` the values?

File: test/Microsoft.ML.Tests/OnnxConversionTest.cs [hunk 2]
```diff
@@ -846,6 +847,71 @@ void IndicateMissingValuesOnnxConversionTest()
             Done();
         }
 
+        private class TextData
+        {
+            public string Text { get; set; }
+        }
+
+        [Fact]
+        void NgramOnnxConnversionTest()
```
Comment (yaeldekel) [inline test/Microsoft.ML.Tests/OnnxConversionTest.cs]: >NgramOnnxConnversionTest [](start = 13, length = 24)

Can you please add tests for WordTokenizing and ValueToKeyMapping as well?

Comment (harishsk) [reply test/Microsoft.ML.Tests/OnnxConversionTest.cs]: The pipeline below already covers WordTokenizing, ValueToKeyMapping and NgramTransforms. Did you mean separate tests for each of the transforms?

---
In reply to: [333957174](https://github.com/dotnet/machinelearning/pull/4312#discussion_r333957174) [](ancestors = 333957174)

**Comment by harishsk** [comment]
This PR has a lot of unnecessary history from merges that is causing problems down the line. So I am closing this PR and opening a new draft PR with just the necessary commits.

**Pull request closed without merging**